public/app/plugins/panel/timeseries/utils.ts

```diff
@@ -173,3 +178,24 @@ export function regenerateLinksSupplier(
 
   return alignedDataFrame;
 }
+
+export function prepareTimeSeriesLong(series: DataFrame[]): DataFrame[] {
+  const stringFields = series[0].fields.filter((field) => field.type === 'string').map((field) => field.name);
+
+  const options = {
+    fields: stringFields,
+    naming: {
+      asLabels: true,
+      append: false,
+      withNames: false,
+      separator1: '=',
+      separator2: ' ',
+    },
+  };
+
+  const ctx = {
+    interpolate: (value: string) => value,
+  };
+
+  return partitionByValuesTransformer.transformer(options, ctx)(series);
```

Comment:
if you move the options inline here, vscode should give you autocomplete :) as a const struct it can not help you much (unless it is invalid)